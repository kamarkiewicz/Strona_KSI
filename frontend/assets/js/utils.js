
// Access image field from backend
function getImage (image) {
  if ((image || {}).path) {
    // path may be relative URL
    let isAbsolute = /^(?:[a-z]+:)?\/\//i.test(image.path)
    image.path = (isAbsolute ? '' : process.env.API_URL) + image.path
  }
  return image
}

export { getImage };
