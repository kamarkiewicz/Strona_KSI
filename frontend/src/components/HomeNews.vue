<template>
  <div class="home-news py-5">
    <div class="container">
      <div class="row">
        <div class="col-md-12">
          <h2 class="text-primary">{{ title }}</h2>
        </div>
      </div>
      <div class="row">
        <div class="col-md-12">
          <div class="row">

            <div class="col-md-4" v-for="news in newsList" :key="news.id">
              <img class="img-fluid d-block w-100 mb-3 rounded" :src="news.pic">
              <h2 class="text-primary">{{ news.title }}</h2>
              <p class="">{{ news.content }}</p>
            </div>

          </div>
        </div>
      </div>
    </div>
  </div>
</template>

<script>
export default {
  name: 'HomeNewsComponent',
  data () {
    return {
      title: 'Aktualności',
      newsList: [
        { id: 1,
          pic: 'https://picsum.photos/270/180?image=1024',
          title: 'Tytuł aktualności 1',
          pubDate: 'wczoraj',
          content: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.'
        },
        { id: 2,
          pic: 'https://picsum.photos/270/180?image=1020',
          title: 'Tytuł aktualności 2',
          pubDate: '2 dni temu',
          content: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.'
        },
        { id: 3,
          pic: 'https://picsum.photos/270/180?image=1010',
          title: 'Tytuł aktualności 3',
          pubDate: '1 czerwca 2001',
          content: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.'
        }
      ]
    }
  }
}
</script>
