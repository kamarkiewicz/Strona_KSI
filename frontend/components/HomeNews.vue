<template>
  <section class="home-news bg-light">

    <header class="container pt-5 pb-4">
      <h2 class="text-primary" v-text="$t('general.news')"></h2>
    </header>

    <div class="container px-sm-5">
      <div class="row equal">
        <div class="col-lg-4 pr-3 pb-5" v-for="el in news" :key="el.id">
          <preview-card
            :image="el.image"
            :date="el.date"
            :title="el.title"
            :excerpt="el.excerpt"
            :link="localePath({ name: 'news-article', params: { article: el.link }})"
            linkText="Czytaj wiÄ™cej" />
        </div>
      </div>
    </div>

  </section>
</template>

<script>
import { Image } from '~/assets/js/utils'
import PreviewCard from '~/components/PreviewCard.vue'

export default {
  components: {
    PreviewCard,
  },
  props: {
    news: {
      type: Array, // must contain max 3 entries
      required: true
    },
  },
}
</script>
