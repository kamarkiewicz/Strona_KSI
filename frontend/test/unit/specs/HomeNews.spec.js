import Vue from 'vue'
import { mount } from '@vue/test-utils'
import HomeNews from '~/components/HomeNews.vue'
import { Image } from '~/assets/js/utils'

Vue.config.ignoredElements = [
  'b-img', 'b-img-lazy', 'nuxt-link'
];

describe('HomeNews', () => {
  let wrapper

  beforeEach(() => {
    wrapper = mount(HomeNews, {
      mocks: {
        $t: (label) => { return label },
        $d: (label) => { return label },
        localePath: (args) => { return JSON.stringify(args) },
      },
      propsData: {
        news: [
          {
            id: 1,
            date: 1527628516335,
            title: 'Tytuł aktualności 1',
            excerpt: '<p>A party you cannot refuse...</p><p>Osoby zainteresowane proszone są o imienny zapis w pokoju KSI.</p>',
            image: new Image('https://picsum.photos/270/180?image=1024'),
            link: 'tytul-aktualnosci-1'
          },
          {
            id: 2,
            date: 1527628516335,
            title: 'Tytuł aktualności 2',
            excerpt: '<p>A party you cannot refuse...</p><p>Osoby zainteresowane proszone są o imienny zapis w pokoju KSI.</p>',
            image: new Image('https://picsum.photos/270/180?image=1020'),
            link: 'tytul-aktualnosci-2'
          },
          {
            id: 3,
            date: 1527628516335,
            title: 'Tytuł aktualności 3',
            excerpt: '<p>A party you cannot refuse...</p><p>Osoby zainteresowane proszone są o imienny zapis w pokoju KSI.</p>',
            image: new Image('https://picsum.photos/270/180?image=1010'),
            link: 'tytul-aktualnosci-3'
          },
        ],
      },
    })
  })

  it('renders correctly', () => {
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('contains News header', () => {
    const label = 'general.news'
    expect(wrapper.text()).toMatch(label)
  })
})
