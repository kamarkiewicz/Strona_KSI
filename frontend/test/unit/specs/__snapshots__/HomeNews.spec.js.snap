// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeNews renders correctly 1`] = `
<section class="home-news bg-light">
  <header class="container pt-5 pb-4">
    <h2 class="text-primary">general.news</h2>
  </header>
  <div class="container px-sm-5">
    <div class="row equal">
      <div class="col-lg-4 pr-3 pb-5">
        <article class="preview-card card h-100 rounded border-0">
          <b-img-lazy src="https://picsum.photos/270/180?image=1024" alt="Featured image" class="card-img-top rounded-top"></b-img-lazy>
          <div class="card-body">
            <header>
              <h6 class="card-subtitle mb-2 text-muted">1527628516335</h6>
              <h5 class="card-title">Tytuł aktualności 1</h5>
            </header>
            <div class="card-text text-justify">
              <p>A party you cannot refuse...</p>
              <p>Osoby zainteresowane proszone są o imienny zapis w pokoju KSI.</p>
            </div>
            <nuxt-link to="{&quot;name&quot;:&quot;news-article&quot;,&quot;params&quot;:{&quot;article&quot;:&quot;tytul-aktualnosci-1&quot;}}" class="card-link pretty-link">general.readMore</nuxt-link>
          </div>
        </article>
      </div>
      <div class="col-lg-4 pr-3 pb-5">
        <article class="preview-card card h-100 rounded border-0">
          <b-img-lazy src="https://picsum.photos/270/180?image=1020" alt="Featured image" class="card-img-top rounded-top"></b-img-lazy>
          <div class="card-body">
            <header>
              <h6 class="card-subtitle mb-2 text-muted">1527628516335</h6>
              <h5 class="card-title">Tytuł aktualności 2</h5>
            </header>
            <div class="card-text text-justify">
              <p>A party you cannot refuse...</p>
              <p>Osoby zainteresowane proszone są o imienny zapis w pokoju KSI.</p>
            </div>
            <nuxt-link to="{&quot;name&quot;:&quot;news-article&quot;,&quot;params&quot;:{&quot;article&quot;:&quot;tytul-aktualnosci-2&quot;}}" class="card-link pretty-link">general.readMore</nuxt-link>
          </div>
        </article>
      </div>
      <div class="col-lg-4 pr-3 pb-5">
        <article class="preview-card card h-100 rounded border-0">
          <b-img-lazy src="https://picsum.photos/270/180?image=1010" alt="Featured image" class="card-img-top rounded-top"></b-img-lazy>
          <div class="card-body">
            <header>
              <h6 class="card-subtitle mb-2 text-muted">1527628516335</h6>
              <h5 class="card-title">Tytuł aktualności 3</h5>
            </header>
            <div class="card-text text-justify">
              <p>A party you cannot refuse...</p>
              <p>Osoby zainteresowane proszone są o imienny zapis w pokoju KSI.</p>
            </div>
            <nuxt-link to="{&quot;name&quot;:&quot;news-article&quot;,&quot;params&quot;:{&quot;article&quot;:&quot;tytul-aktualnosci-3&quot;}}" class="card-link pretty-link">general.readMore</nuxt-link>
          </div>
        </article>
      </div>
    </div>
  </div>
</section>
`;
